### PR TITLE
Allow use as a meson subproject

### PR DESCRIPTION
**Pull request by jtojnar**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -89,12 +89,12 @@ test_link_args = [
   '-Wl,-z,relro',
   '-Wl,-z,now',
 ]
-foreach arg: test_link_args
-  if cc.has_link_argument(arg)
-    global_link_args += arg
+foreach link_arg: test_link_args
+  if cc.links('int main () { return 0; }', name: link_arg, args: link_arg)
```
Comment (hughsie) [inline meson.build]: is this a workaround? Seems a little hacky...

Comment (jtojnar) [reply meson.build]: You are right, this is a hack, I did not notice has_link_argument was added in 0.45.

**Comment by hughsie** [comment]
@jtojnar the self tests are using non-exported functions, hence why I was building them "again" as part of the  xb-self-test binary.

**Comment by hughsie** [comment]
I've tweaked this, and merged it. I do get some pretty alarming warnings however:

    src/libxmlb/meson.build:5: WARNING: Got an empty configuration_data() object and found no substitutions in the input file 'xb-builder.h'. If you want to copy a file to the build dir, use the 'copy:' keyword argument added in 0.47.0
    WARNING: Output file "src/libxmlb/@PLAINNAME@" for configure_file() at src/libxmlb/meson.build:5 overwrites configure_file() output at src/libxmlb/meson.build:5
    Configuring xb-builder-node.h using configuration
    src/libxmlb/meson.build:5: WARNING: Got an empty configuration_data() object and found no substitutions in the input file 'xb-builder-node.h'. If you want to copy a file to the build dir, use the 'copy:' keyword argument added in 0.47.0
    WARNING: Output file "src/libxmlb/@PLAINNAME@" for configure_file() at src/libxmlb/meson.build:5 overwrites configure_file() output at src/libxmlb/meson.build:5
    Configuring xb-builder-source.h using configuration
    src/libxmlb/meson.build:5: WARNING: Got an empty configuration_data() object and found no substitutions in the input file 'xb-builder-source.h'. If you want to copy a file to the build dir, use the 'copy:' keyword argument added in 0.47.0

**Comment by jtojnar** [comment]
Yes, that is expected. We can change it to use copy kwarg once we upgrade to meson 0.47.

**Comment by hughsie** [comment]
@jtojnar  -- I think we can do that now; Centos even has 0.47.2 -- can you do a PR with that change please? Thanks!

**Comment by jtojnar** [comment]
I pushed one more fix to the subproject branch. And opened PR about the `copy` kwarg #2.

**Pull request closed without merging**